src/api/app/controllers/staging_project/staged_requests_controller.rb

```diff
@@ -1,5 +1,59 @@
 class StagingProject::StagedRequestsController < StagingProjectController
+  before_action :set_staging_workflow, :set_project, :set_xml_hash, only: [:create, :destroy]
+
   def index
-    @requests = @project.staged_requests
+    @requests = @staging_project.staged_requests
+  end
+
+  def create
+    authorize @staging_project, :update?
+    requests = @staging_workflow.unassigned_requests.where(id: @xml_hash[:number])
+
+    if @staging_project.staged_requests << requests
```

Comment:
Am I looking at the wrong [documentation](https://api.rubyonrails.org/classes/ActiveRecord/Associations/CollectionProxy.html#method-i-3C-3C) or does this not work as expected? :-)

Reply:
> Am I looking at the wrong [documentation](https://api.rubyonrails.org/classes/ActiveRecord/Associations/CollectionProxy.html#method-i-3C-3C) or does this not work as expected?

what is exactly the necessary change here?

Reply:
IMO: try to cover the render_error and we'll see

Reply:
> what is exactly the necessary change here?

" <<(*records)

Adds one or more records to the collection by setting their foreign keys to the association's primary key. Returns self, so several appends may be chained together."

Sounds like it always returns itself, the collection proxy, doesn't it?

Reply:
Yeah, I think it will always be true probably. My bad. But also not sure what can possible go wrong with adding the assocation (famous last words...) so we might drop the else part?

Reply:
I'm fine with dropping the else for now but I also don't like this solution. For instance a users POSTS requests 1 & 2 but one request does not exist. In this reason we should give the user feedback about this fact. Of course this needs to get implemented in a different way then.

Something like:

```
@staging_project.staged_requests << requests
missing_requests = (requests - @hash[:number])

if missing_requests.empty?
  # render ok, no requests missing
else
  # render error
end
```

Out of scope for this PR but some food for thoughts...

Reply:
everything which depends on the the results from the background job won't be shown/catched there unless you block. Normally, what can be do in the API is provide an endpoint to ask for the job status (which most queue provide out-of-the-box).

Reply:
I agree with @ChrisBr. It can also be done by checking that the number of records returned by the `where` is the same as expected. That is more efficient than the difference of arrays.

Reply:
> ChrisBr a day ago 
> I'm fine with dropping the else for now but I also don't like this solution. For instance a users POSTS requests 1 & 2 but one request does not exist. In this reason we should give the user feedback about this fact.

That sounds like a good idea :thumbsup:

Reply:
> everything which depends on the the results from the background job won't be shown/catched there unless you block. Normally, what can be do in the API is provide an endpoint to ask for the job status (which most queue provide out-of-the-box).

@vpereira this does not depend on the background job, this only depends on the assocations so it does not block.